src/NHibernate/Dialect/SybaseSQLAnywhere10Dialect.cs

```diff
@@ -343,6 +343,8 @@ protected virtual void RegisterMiscellaneousFunctions()
 			RegisterFunction("transactsql", new StandardSQLFunction("transactsql", NHibernateUtil.String));
 			RegisterFunction("varexists", new StandardSQLFunction("varexists", NHibernateUtil.Int32));
 			RegisterFunction("watcomsql", new StandardSQLFunction("watcomsql", NHibernateUtil.String));
+			RegisterFunction("truncnum", new StandardSafeSQLFunction("truncnum", 2));
```

Comment:
I do not think that we should register specific function names for being available in HQL when using their specific dialect, since HQL should be database agnostic. `RegisterFunction("truncnum"` should not have been added.

Reply:
I think opposite. Moreover, this is how it always was: dialect specific functions + aliases to the more common name.

Reply:
"Always" widely depends on dialect. Some do register specific names for some of their functions in addition to the generic one, others do not. (By example SQL-Server registers only generic names for `getdate`, `len`, `datepart`, `charindex`, same for Postgres and `now`, ...)
It looks to me as something quite accidental.

This practice of registering specific names does not promote portability of HQL, while an ORM is usually meant to be database agnostic. That is why I think this is a bad thing, causing me to prefer not adding new specific names when we have a generic one. (Removing them after a release would of course be a breaking change for those using them.)